ostd/src/mm/vm_space.rs

```diff
@@ -450,6 +427,73 @@ impl CursorMut<'_, '_> {
     }
 }
 
+impl<'a, 'b> CursorMut<'a, 'b, false> {
+    /// Upgrade the cursor to a mutable cursor that can unmap memory.
+    pub fn upgrade(mut self) -> CursorMut<'a, 'b, true> {
+        let guard = self.activation_lock.read();
+        self.flusher.update(self.cpus.load());
```

Comment:
Good job! This PR makes the lock used for TLB coherence more flexible, but I don't feel it's necessary to distinguish unmap cursor. I think we could incorporate the `upgrade` logic within the unmap function and handle the scenario where there is an existing old mapping, ensuring TLB coherence during this period. 

This approach might slightly impact the performance of consecutive unmaps, but I believe such scenarios are not frequent (I'm not sure). Additionally, it would allow a cursor to continue mapping after completing an unmap (since there is currently no downgrade method).